R/git.R

```diff
@@ -323,85 +323,117 @@ git_clean <- function() {
-  hd_line("Git repo for current project")
+  if ("project" %in% scope) {
 
-  if (!proj_active()) {
-    ui_info("No active usethis project")
-    return(invisible())
-  }
-  kv_line("Active usethis project", proj_get())
-  if (!uses_git()) {
-    ui_info("Active project is not a Git repo")
-    return(invisible())
-  }
+    if (!proj_active()) {
+      ui_info("No active usethis project")
+      return(invisible())
+    }
+    kv_line("Active usethis project", proj_get())
+    if (!uses_git()) {
+      ui_info("Active project is not a Git repo")
+      return(invisible())
+    }
+
+    # current branch -------------------------------------------------------------
+    branch <- tryCatch(git_branch(), error = function(e) NULL)
+    tracking_branch <- if (is.null(branch)) NA_character_ else git_branch_tracking()
+    # TODO: can't really express with kv_line() helper
+    branch <- if (is.null(branch)) "<unset>" else branch
+    tracking_branch <- if (is.na(tracking_branch)) "<unset>" else tracking_branch
+
+    # local git config -----------------------------------------------------------
+    if ("git" %in% tool) {
+      cli::cli_h3("Git (project)")
+      if (proj_active() && uses_git()) {
+        local_user <- list(
+          user.name = git_cfg_get("user.name", "local"),
+          user.email = git_cfg_get("user.email", "local")
+        )
+        if (!is.null(local_user$user.name) || !is.null(local_user$user.email)) {
+          ui_info("This repo has a locally configured user")
+        }
+        if (!is.null(local_user$user.name)) {
+          kv_line("Name", local_user$user.name)
+        }
+        if (!is.null(local_user$user.email)) {
+          kv_line("Email", local_user$user.email)
+        }
+      }
+
+      # default branch -------------------------------------------------------------
+      default_branch_sitrep()
 
-  # local git config -----------------------------------------------------------
-  if (proj_active() && uses_git()) {
-    local_user <- list(
-      user.name = git_cfg_get("user.name", "local"),
-      user.email = git_cfg_get("user.email", "local")
-    )
-    if (!is.null(local_user$user.name) || !is.null(local_user$user.name)) {
-      ui_info("This repo has a locally configured user")
-      kv_line("Name", local_user$user.name)
-      kv_line("Email", local_user$user.email)
+      # vertical alignment would make this nicer, but probably not worth it
+      ui_bullet(glue("
+      Current local branch -> remote tracking branch:
+      {ui_value(branch)} -> {ui_value(tracking_branch)}"))
+    }
+
+    # GitHub remote config -------------------------------------------------------
+    if ("github" %in% tool) {
+      cfg <- github_remote_config()
+      repo_host <- cfg$host_url
+      if (!is.na(repo_host) && repo_host != default_gh_host) {
+        cli::cli_h3("GitHub (project) host")
+        kv_line("Non-default GitHub host", repo_host)
+        pat_sitrep(repo_host, scold_for_renviron = FALSE)
```

Comment:
Perhaps this can be integrated with the remote config. I.e. maybe we only need 1 heading here, not (potentially) two.

Reply:
Perhaps "subheadings" could be used here, which will appear only if the Github host is non-default.

<details>
<summary>Non-default case</summary>

```
── GitHub project 
Host:
• Non-default GitHub host: 'https://github.com'
• Personal access token for 'https://github.com': '<discovered>'
• GitHub user: 'ijlyttle'
• Token scopes: 'gist, repo, user, workflow'
• Email(s): 'ijlyttle@me.com (primary)', 'ijlyttle@users.noreply.github.com', 'ian.lyttle@<work>.com'
Project:
• Type = 'fork'
• Host = 'https://github.com'
• Config supports a pull request = TRUE
• origin = 'ijlyttle/usethis' (can push) = fork of 'r-lib/usethis'
• upstream = 'r-lib/usethis' (can not push)
• Desc = 'origin' is a fork of 'r-lib/usethis', which is configured as the 'upstream' remote.
```
</Details>

<details>
<summary>Default case</summary>

```
── GitHub project 
• Type = 'fork'
• Host = 'https://github.com'
• Config supports a pull request = TRUE
• origin = 'ijlyttle/usethis' (can push) = fork of 'r-lib/usethis'
• upstream = 'r-lib/usethis' (can not push)
• Desc = 'origin' is a fork of 'r-lib/usethis', which is configured as the 'upstream' remote.
```
</Details>

proposed in 08344f95